### PR TITLE
Adding webapp keyvault ssl template

### PR DESCRIPTION
**Pull request by garvinmsft**
### Changelog

* Adding a template for webapps configured with ssl certificates from Keyvault

### CODE REVIEW

File: webapp-keyvault-ssl/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,128 @@
+{
+   "$schema":"https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+   "contentVersion":"1.0.0.0",
+   "variables": {
+    "linkedTemplateBase": "[uri(deployment().properties.templateLink.uri, 'nestedtemplates')]"
```
Comment (bmoore-msft) [inline webapp-keyvault-ssl/azuredeploy.json]: see the BP guide for how to handle artifacts

**Comment by garvinmsft** [comment]
I added a new commit containing best practice recommendations.

File: webapp-keyvault-ssl/azuredeploy.parameters.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{
+   "$schema":"http://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+   "contentVersion":"1.0.0.0",
+   "parameters":{
+      "certificateName":{
+         "value":"GEN-UNIQUE"
+      },
+      "existingKeyVaultId":{
+         "value":"/subscriptions/changeme/resourceGroups/changeme/providers/Microsoft.KeyVault/vaults/GEN-UNIQUE"
```
Comment (bmoore-msft) [inline webapp-keyvault-ssl/azuredeploy.parameters.json]: use:  GEN-KEYVAULT-RESOURCE-ID

File: webapp-keyvault-ssl/azuredeploy.parameters.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+   "contentVersion":"1.0.0.0",
+   "parameters":{
+      "certificateName":{
+         "value":"GEN-UNIQUE"
+      },
+      "existingKeyVaultId":{
+         "value":"/subscriptions/changeme/resourceGroups/changeme/providers/Microsoft.KeyVault/vaults/GEN-UNIQUE"
+      },
+      "existingKeyVaultSecretName":{
+         "value":"[testcert1]"
```
Comment (bmoore-msft) [inline webapp-keyvault-ssl/azuredeploy.parameters.json]: use: GEN-KEYVAULT-SSL-SECRET-NAME

**Comment by garvinmsft** [comment]
Updated!

**Pull request merged**